### PR TITLE
qa/deepsea: make re-invocation of deepsea task re-install DS

### PR DESCRIPTION
**Pull request by smithfarm**
To do:

- [x] test all four permutations (source, source; source, package; package, source; package, package)
- [x] drop `deepsea/test/upgrade`
- [x] drop deepsea-install-from-source validation test (now obsolete)
- [x] run `deepsea/tier4/upgrade` providing SES6 repos via override
- [x] run `deepsea/tier4/upgrade` providing SES6 repos _and_ `upgrade_repo`+`upgrade_branch` via override (the latter should trigger reinstall from source)
- [x] use this PR to test https://github.com/SUSE/DeepSea/pull/1509
- [x] move dummy[1234] to `deepsea/tier4/deepsea-reinstall` (or something similar)
- [x] approval
- [x] merge
- [ ] forward-port to ses6

### CODE REVIEW

**Comment by smithfarm** [comment]
there's a problem with this . . . current form does not allow to specify a deepsea repo/branch just for the "upgrade" via override, because override will affect both invocations of "deepsea" task

**Comment by smithfarm** [comment]
@jschmid1 I have tested this with the following result:

* it works as advertised (re-invoking the deepsea task really does reinstall deepsea in the manner indicated)
* it does not seem to cause any regression (I ran `--suite deepsea/tier0` successfully using this branch)

Ready for review?

**Comment by kshtsk** [comment]
retest this please

**Comment by smithfarm** [comment]
@kshtsk Note, this is SES5, dependencies should be OK.

**Comment by kshtsk** [comment]
retest this please

**Comment by kshtsk** [comment]
@smithfarm so what's next?

**Comment by kshtsk** [comment]
retest this please

**Comment by kshtsk** [comment]
retest this please

**Comment by kshtsk** [comment]
@smithfarm The leap 42.3 does not have the python-imagesize dependency issue: http://storage-ci.suse.de:8080/job/mkck-run/31/console

File: qa/suites/deepsea/tier4/upgrade/3-test-phase.yaml [hunk 1]
```diff
@@ -16,13 +13,23 @@ tasks:
...
+                        - 'deepsea --version || true'
+                        - 'salt-run deepsea.version || true'
         - deepsea.repository:
                 all:
         - exec:
                 all:
                         - 'zypper --non-interactive --no-gpg-checks rm gvfs-backends sle-sdk-release-POOL || true'
         - salt.command:
                 command: 'state.apply ceph.updates.salt'
+        - salt.command:
```
Comment (jschmid1) [inline qa/suites/deepsea/tier4/upgrade/3-test-phase.yaml]: salt is only partly installed (due to unresolved vendor changes. weirdly enough this only happens on 2/4 nodes repeatedly)

```
Problem: salt-api-2016.11.4-48.9.33.x86_64 requires salt = 2016.11.4-48.9.33, but this requirement cannot be provided
  deleted providers: salt-2016.11.4-48.9.33.x86_64
 Solution 1: Following actions will be done:
  install salt-api-2019.2.0rc1-5.2.x86_64 (with vendor change)
    obs://build.suse.de/Devel:Storage:5.0  -->  SUSE LLC <https://www.suse.com/>
  install salt-master-2019.2.0rc1-5.2.x86_64 (with vendor change)
    obs://build.suse.de/Devel:Storage:5.0  -->  SUSE LLC <https://www.suse.com/>
 Solution 2: do not install salt-2019.2.0rc1-5.2.x86_64
 Solution 3: break salt-api-2016.11.4-48.9.33.x86_64 by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/3/c] (c): y
Invalid answer 'y'. [1/2/3/c] (c): 1
Resolving dependencies...
Resolving package dependencies...
```

File: qa/suites/deepsea/tier4/upgrade/3-test-phase.yaml [hunk 2]
```diff
@@ -26,6 +26,10 @@ tasks:
...
+                command: 'cmd.shell zypper -n in -f --from ses --force-resolution salt'
+        - salt.command:
+                command: 'cmd.shell rpm -qa | grep salt'
         - salt.command:
                 command: 'saltutil.sync_all'
```
Comment (smithfarm) [inline qa/suites/deepsea/tier4/upgrade/3-test-phase.yaml]: Hm, I wonder, would it make sense to let `salt.command` optionally take an array of strings? Then all four of the above invocations could be folded into a single invocation.

**Comment by smithfarm** [comment]
@jschmid1 Is this ready to merge?

File: qa/suites/deepsea/tier4/upgrade/3-test-phase.yaml (hunk 2 above)
Comment (jschmid1) [reply qa/suites/deepsea/tier4/upgrade/3-test-phase.yaml]: Mhh, I don't mind keeping things explicit..

**Comment by jschmid1** [comment]
> @jschmid1 Is this ready to merge?

yeah, let's merge this. 

maybe my fix `'cmd.shell zypper -n in -f --from ses --force-resolution salt'` has to be reverted in the future.. 

lgtm for now

**Comment by smithfarm** [comment]
@jschmid1 OK, I left your commit out of the ses6 forwardport.

**Pull request merged**